src/main/java/racingcar/domain/car/CarInfo.java

```diff
@@ -0,0 +1,51 @@
+package racingcar.domain.car;
+
+import static racingcar.constant.GameConstant.INITIAL_CAR_INDEX;
+import static racingcar.exception.InputException.existsNameException;
+import static racingcar.exception.InputException.notExistsNameException;
+
+import java.util.List;
+import java.util.stream.IntStream;
+import racingcar.dto.Car;
+import racingcar.repository.CarRepository;
+
+public class CarInfo {
+    private static final CarRepository carRepository = CarRepository.getInstance();
+    private static final CarInfo instance = new CarInfo();
+
+    public static CarInfo getInstance() {
+        return instance;
+    }
+
+    public void saveCarInfo(Car car) {
+        if (carRepository.hasName(car.name())) {
+            existsNameException();
+        }
+        carRepository.saveCar(car);
+    }
+
+    public void saveAllCarInfo(List<String> carList) {
+        IntStream.range(0, carList.size())
+                .mapToObj(id -> new Car(carList.get(id), INITIAL_CAR_INDEX))
+                .forEach(this::saveCarInfo);
+    }
```

Comment:
IntStream보다는 carList를 활용하는게 더 좋아보여요!
```suggestion
    public void saveAllCarInfo(List<String> carList) {
           carList.stream()
                .map(name -> new Car(name, INITIAL_CAR_INDEX)
                .forEach(this::saveCarInfo);
    }
```